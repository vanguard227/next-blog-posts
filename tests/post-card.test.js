import renderer from "react-test-renderer";
import PostCard from "../components/post-card";
import { shallow } from "enzyme";

describe("<PostCard />", () => {
  const title = "Title1";
  const createdAt = "2020-10-07T21:01:51.705Z";
  const description = "Post description";
  const authors = [
    {
      id: "1",
      name: "John Smith",
      avatar: "https://cdn.fakercloud.com/avatars/iamsteffen_128.jpg",
    },
  ];
  const comments = [
    {
      id: "1",
      title: "Comment1",
      description: "Comment description",
      createdAt: "2020-10-07T21:01:51.705Z",
    },
  ];

  it("matches snapshot", () => {
    const component = renderer.create(
      <PostCard
        title={title}
        createdAt={createdAt}
        description={description}
        authors={authors}
        comments={comments}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders PostCard", () => {
    const wrapper = shallow(
      <PostCard
        title={title}
        createdAt={createdAt}
        description={description}
        authors={authors}
        comments={comments}
      />
    );

    let actual = wrapper.find("h3").text();
    expect(actual).toBe(title);

    actual = wrapper.find("span").text();
    const createdTime = new Date(createdAt).toLocaleDateString();
    expect(actual).toBe(createdTime);

    actual = wrapper.find("p").text();
    expect(actual).toBe(description);

    actual = wrapper.find("button").text();
    expect(actual).toBe(`Comments (${comments.length})`);
  });

  it("click comments button", () => {
    const wrapper = shallow(
      <PostCard
        title={title}
        createdAt={createdAt}
        description={description}
        authors={authors}
        comments={comments}
      />
    );

    let actual = wrapper.find("Comments").length;
    expect(actual).toBe(0);

    const commentBtn = wrapper.find('button');
    commentBtn.simulate('click');

    actual = wrapper.find("Comments").length;
    expect(actual).toBe(1);
  });
});
