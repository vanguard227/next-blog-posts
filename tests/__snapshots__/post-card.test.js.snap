// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> matches snapshot 1`] = `
<div
  className="sc-bczRLJ htlnsA"
>
  <h3>
    Title1
  </h3>
  <span>
    10/7/2020
  </span>
  <p>
    Post description
  </p>
  <div
    className="sc-dkzDqf ffKjMI"
  >
    <div>
      <div
        className="sc-hKMtZM bjFhZq"
      >
        <img
          alt="image"
          src="https://i.pinimg.com/564x/64/a5/f2/64a5f243aa5efbea4bb704007459f4c3--create-a-cartoon-annoying-things.jpg"
        />
      </div>
      <p>
        John Smith
      </p>
    </div>
  </div>
  <button
    className="sc-gsnTZi bioVKy"
    onClick={[Function]}
  >
    Comments (
    1
    )
  </button>
</div>
`;
